### PR TITLE
Change DS config items to dsmap_t

### LINKED ISSUE #4200: allow multiple trust-anchors for the same name in the recursor
every next addDS() is replacing the values of the previous one. Multiple anchors are necessary during (root) key/algorithm rollovers.

Proposed format
```
-- remove default key
clearDS()
-- add new keys
addDS(".", "first.......")
addDS(".", "next........")
~~~

And maybe add a dnskey2DS() function so it is no longer necessary to convert DNSKEYs with external tools like ldns-key2ds in DS record format.
```
addDS(".", dnskey2DS("...."))
```

### PR DESCRIPTION
**Pull request by pieterlexis**
Ensure that addTA() appends the DS.

Closes  #4200 

This does have some code duplication though....

### CODE REVIEW

**Comment by Habbie** [comment]
I don't see any deduplication of DSes, am I missing something?

File: docs/manpages/rec_control.1.md [hunk 1]
```diff
@@ -47,7 +47,8 @@ add-nta *DOMAIN* [*REASON*]
 :    Add a Negative Trust Anchor for *DOMAIN*, suffixed optionally with *REASON*.
 
 add-ta *DOMAIN* *DSRECORD*
-:    Add a Trust Anchor for *DOMAIN* with DS record data *DSRECORD*.
+:    Add a Trust Anchor for *DOMAIN* with DS record data *DSRECORD*. This does not
+     replace the existing Trust Anchor for *DOMAIN* but appends the *DSRECORD*.
```
Comment (Habbie) [inline docs/manpages/rec_control.1.md]: This text comes across a bit vague. Maybe 'This adds the new TA to the existing TA set for the domain.' or the like

File: pdns/toysdig.cc [hunk 2]
```diff
@@ -101,8 +101,7 @@ GlobalStateHolder<LuaConfigItems> g_luaconfs;
 LuaConfigItems::LuaConfigItems()
 {
   auto ds=std::unique_ptr<DSRecordContent>(dynamic_cast<DSRecordContent*>(DSRecordContent::make("19036 8 2 49aac11d7b6f6446702e54a1607371607a1a41855200fd2ce1cdde32f24e8fb5")));
```
Comment (Habbie) [inline pdns/toysdig.cc]: not this PRs fault but we should really move 19036 into a central place.

Comment (pieterlexis) [inline pdns/toysdig.cc]: pdns/root_addresses.hh seems appropriate. This should be a new ticket

**Comment by Habbie** [comment]
LGTM

**Pull request merged**